src/algebra/group/type_tags.lean

```diff
@@ -166,21 +161,32 @@ instance [comm_group α] : add_comm_group (additive α) :=
 instance [add_comm_group α] : comm_group (multiplicative α) :=
 { .. multiplicative.group, .. multiplicative.comm_monoid }
 
-/-- Reinterpret `f : α →+ β` as `multiplicative α →* multiplicative β`. -/
-def add_monoid_hom.to_multiplicative [add_monoid α] [add_monoid β] (f : α →+ β) :
-  multiplicative α →* multiplicative β :=
-⟨f.1, f.2, f.3⟩
-
-/-- Reinterpret `f : α →* β` as `additive α →+ additive β`. -/
-def monoid_hom.to_additive [monoid α] [monoid β] (f : α →* β) : additive α →+ additive β :=
-⟨f.1, f.2, f.3⟩
-
-/-- Reinterpret `f : additive α →+ β` as `α →* multiplicative β`. -/
-def add_monoid_hom.to_multiplicative' [monoid α] [add_monoid β] (f : additive α →+ β) :
-  α →* multiplicative β :=
-⟨f.1, f.2, f.3⟩
-
-/-- Reinterpret `f : α →* multiplicative β` as `additive α →+ β`. -/
-def monoid_hom.to_additive' [monoid α] [add_monoid β] (f : α →* multiplicative β) :
-  additive α →+ β :=
-⟨f.1, f.2, f.3⟩
+/-- Reinterpret `α →+ β` as `multiplicative α →* multiplicative β`. -/
+def add_monoid_hom.to_multiplicative [add_monoid α] [add_monoid β] :
+  (α →+ β) ≃ (multiplicative α →* multiplicative β) :=
+⟨λ f, ⟨f.1, f.2, f.3⟩, λ f, ⟨f.1, f.2, f.3⟩, λ x, by { ext, refl, }, λ x, by { ext, refl, }⟩
+
+/-- Reinterpret `α →* β` as `additive α →+ additive β`. -/
+def monoid_hom.to_additive [monoid α] [monoid β] :
+  (α →* β) ≃ (additive α →+ additive β) :=
+⟨λ f, ⟨f.1, f.2, f.3⟩, λ f, ⟨f.1, f.2, f.3⟩, λ x, by { ext, refl, }, λ x, by { ext, refl, }⟩
+
+/-- Reinterpret `additive α →+ β` as `α →* multiplicative β`. -/
+def add_monoid_hom.to_multiplicative' [monoid α] [add_monoid β] :
+  (additive α →+ β) ≃ (α →* multiplicative β) :=
+⟨λ f, ⟨f.1, f.2, f.3⟩, λ f, ⟨f.1, f.2, f.3⟩, λ x, by { ext, refl, }, λ x, by { ext, refl, }⟩
+
+/-- Reinterpret `α →* multiplicative β` as `additive α →+ β`. -/
+def monoid_hom.to_additive' [monoid α] [add_monoid β] :
+  (α →* multiplicative β) ≃ (additive α →+ β) :=
+add_monoid_hom.to_multiplicative'.symm
+
+/-- Reinterpret `α →+ additive β` as `multiplicative α →* β`. -/
+def add_monoid_hom.to_multiplicative'' [add_monoid α] [monoid β] :
+  (α →+ additive β) ≃ (multiplicative α →* β) :=
+⟨λ f, ⟨f.1, f.2, f.3⟩, λ f, ⟨f.1, f.2, f.3⟩, λ x, by { ext, refl, }, λ x, by { ext, refl, }⟩
```

Comment:
A lot of the `by { ext, refl, }`s could be replaced with `add_monoid_hom.ext (λ _, rfl)` but I don't know if it is worth it.

Reply:
I'm happy with it either way. The tactic proof omits a little bit of noise, since it doesn't need the `add_monoid_hom` qualification. Since it's such a short proof, that's not really important though.

Reply:
I'll leave it as is then.